### PR TITLE
feat: timeline view

### PR DESCRIPTION
**Pull request by csirius**
# TL;DR
Add timeline view for execution details page.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
MVP for timeline view.

## Tracking Issue

Partial work for: https://github.com/flyteorg/flyte/issues/1925

## Follow-up issue
_NA_

### CODE REVIEW

File: src/components/Executions/ExecutionDetails/Timeline/ExecutionTimeline.tsx [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+import { makeWorkflowQuery } from 'components/Workflow/workflowQueries';
+import { Workflow, WorkflowClosure } from 'models/Workflow/types';
+import { useQuery, useQueryClient } from 'react-query';
+import { WaitForQuery } from 'components/common/WaitForQuery';
+import { DataError } from 'components/Errors/DataError';
+import { dNode } from 'models/Graph/types';
+import { transformerWorkflowToDAG } from 'components/WorkflowGraph/transformerWorkflowToDAG';
+import { isEndNode, isStartNode } from 'components/WorkflowGraph/utils';
+
+ChartJS.register(...registerables, ChartDataLabels);
```
Comment (anrusina) [inline src/components/Executions/ExecutionDetails/Timeline/ExecutionTimeline.tsx:28]: nit: add comment why this register is needed (for future devs)

File: src/components/Executions/ExecutionDetails/Timeline/ExecutionTimelineFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    root: {
+        color: '#665AFF',
+        height: 4,
+        padding: '15px 0',
+        width: 360
+    },
+    active: {},
+    valueLabel: {
+        left: 'calc(-50% + 12px)',
+        color: '#000',
```
Comment (anrusina) [inline src/components/Executions/ExecutionDetails/Timeline/ExecutionTimelineFooter.tsx:42]: nit: here and below `COLOR_SPECTRUM.black`

**Review by anrusina — DISMISSED** [review]
Lgtm. 
- Performed basic smoke test on already finished executions
- All **nit**s, are recommended but not mandatory changes.

File: assetsTransformer.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const path = require('path');
```
Comment (anrusina) [inline assetsTransformer.js:1]: ⚠️ Note: This and file below are also present in Steven's review, which means that they will create a merge conflict for each other.

File: src/components/Executions/ExecutionDetails/Timeline/ExecutionTimeline.tsx [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+                    startedAt.getTime() +
+                    (exec.closure.duration
+                        ? durationToMilliseconds(exec.closure.duration)
+                        : Date.now() - startedTime);
+                if (absoluteDuration > undefinedStart) {
+                    undefinedStart = absoluteDuration;
+                }
+            }
+        }
+        undefinedStart = undefinedStart / 1000;
```
[the remark this answers is not in the document]
Comment (csirius) [reply src/components/Executions/ExecutionDetails/Timeline/ExecutionTimeline.tsx:300]: In my opinion, 

- Adding `/1000` to every calculation is not a good idea as it's a useless computation.
- And `undefinedStart` will not be set only once as it holds the latest start time of the current finished or running executions.
- Third option is also includes useless computation.

[the remark this answers is not in the document]
Comment (anrusina) [reply src/components/Executions/ExecutionDetails/Timeline/ExecutionTimeline.tsx:302]: You right about #2.
Still don't like that think, but see why you don't want to change it. Let it be :)

**Review by anrusina — APPROVED** [review]
From code perspective looks good, haven't done smoke test on the latest code

**Review by anrusina — APPROVED** [review]

**Review by anrusina — APPROVED** [review]

**Pull request merged**